src/_pytest/config/findpaths.py

```diff
@@ -176,7 +176,11 @@ def determine_setup(
         inipath: Optional[Path] = inipath_
         inicfg = load_config_dict_from_file(inipath_) or {}
         if rootdir_cmd_arg is None:
-            rootdir = get_common_ancestor(dirs)
```

Comment:
Actually if ``inifile`` is given, we should use its directory unconditionally as ``rootdir``, so this line should be:

```python
if rootdir_cmd_arg is None:
    rootdir = inipath.parent
```

Reply:
Thanks totally agree